library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/DispatcherLocationProvider.java

```diff
@@ -165,6 +165,31 @@ public void onCancel(DialogInterface dialog) {
               });
 
         if (gpServicesDialog != null) {
+
+            /* Or
+            Intent intent = GoogleApiAvailability.getInstance().getErrorResolutionIntent(getActivity(), gpServicesAvailability, "d");
+
+            if (intent == null)
+            */
+            switch (gpServicesAvailability) {
+                case ConnectionResult.SERVICE_MISSING:
+                case ConnectionResult.SERVICE_VERSION_UPDATE_REQUIRED:
+                case ConnectionResult.SERVICE_DISABLED:
```

Comment:
Let's not do exclusive, but add onDismissListener for the Result type that needs it.
That way is more intentional, and if tomorrow they introduce new types we wouldn't be adding dismiss listener without intentionally. 

```
// With a nice comment that explains this specific case why we need onDismissListener in addition to onCancelListener 
if(gpServicesAvailability == ConnectionResult.SERVICE_INVALID) {
    gpServicesDialog.setOnDismissListener(...)
}
```